bin/copyRawNanoporeDataToPrm.sh

```diff
@@ -0,0 +1,298 @@
+log4Bash 'DEBUG' "${LINENO}" "${FUNCNAME:-main}" '0' "Log files will be written to ${PRM_ROOT_DIR}/logs ..."
+
+#
+# Use multiplexing to reduce the amount of SSH connections created
+# when rsyncing using the group's data manager account.
+# 
+#  1. Become the "${DATA_MANAGER} user who will rsync the data to prm and 
+#  2. Add to ~/.ssh/config:
+#		ControlMaster auto
+#		ControlPath ~/.ssh/tmp/%h_%p_%r
+#		ControlPersist 5m
+#  3. Create ~/.ssh/tmp dir:
+#		mkdir -p -m 700 ~/.ssh/tmp
+#  4. Recursively restrict access to the ~/.ssh dir to allow only the owner/user:
+#		chmod -R go-rwx ~/.ssh
+#
+
+REMOTE_USER="grid"
+REMOTE_MACHINE="145.39.79.24"
+PRM_SAMPLESHEET_DIR="${PRM_ROOT_DIR}/nanopore/Samplesheets"
+SOURCE_DIR="/data/Diagnostiek/"
+DESTINATION_DIR="${PRM_ROOT_DIR}/rawdata/nanopore/"
+
+#
+# Get a list of all samplesheets for this group on the specified samplesheetsServerLocation, where the raw data was generated, and
+#	1. Loop over their analysis ("run") sub dirs and check if there are any we need to rsync.
+#	2. Optionally, split the samplesheets per project after the data was rsynced.
+#
+declare -a runs
+
+# Parse nanopore folder.
+#
+#mapfile -t runs < <(find "${PRM_ROOT_DIR}/rawdata/nanopore/" -maxdepth 1 -mindepth 1 -type d)
+readarray -t runs < <(ssh "${REMOTE_USER}"@"${REMOTE_MACHINE}" "find \"${SOURCE_DIR}\" -maxdepth 1 -mindepth 1 -type d")
+
+for run in "${runs[@]}"
+do
+	#
+	# Process this run
+	#
+	runName="$(basename "${run}")"
+	controlFileBase="${PRM_ROOT_DIR}/logs/${runName}/${runName}"
+	export JOB_CONTROLE_FILE_BASE="${controlFileBase}.${SCRIPT_NAME}"
+	# shellcheck disable=SC2174
+	mkdir -m 2770 -p "${PRM_ROOT_DIR}/logs/${runName}/"
+	if [[ -e "${JOB_CONTROLE_FILE_BASE}.finished" ]]
+	then
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Skipping already processed run ${runName}."
+		continue
+	else
+
+		#
+		# Determine whether an rsync is required for this run, which is the case when
+		# raw data production has finished successfully and this copy script has not.
+		#
+		if ssh "${REMOTE_USER}"@"${REMOTE_MACHINE}" test -e "${SOURCE_DIR}/${runName}/${runName}/"*"/sample_sheet_"*".csv"
+		then
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${REMOTE_USER}"@"${REMOTE_MACHINE}:${SOURCE_DIR}/${runName}/${runName}/"*"/sample_sheet_"*".csv present."
+			finished="true"
+		else
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${REMOTE_USER}"@"${REMOTE_MACHINE}:${SOURCE_DIR}/${runName}/${runName}/"*"/sample_sheet_"*".csv absent."
+			continue
+		fi
+
+		if [[ -e "${PRM_SAMPLESHEET_DIR}/${runName}.csv" ]]
+		then
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${PRM_SAMPLESHEET_DIR}/${runName}.csv present."
+			samplesheetPresent="true"
+		else
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${PRM_SAMPLESHEET_DIR}/${runName}.csv absent."
+			continue
+		fi
+
+		if [[ "${finished}" == "true" && "${samplesheetPresent}" == "true" ]]
```

Comment:
eigenlijk kun je hier alleen nog maar komen als beide files bestaan (want als dat niet zo is zeg je continue en ga je al uit de loop :) )